### DISCUSSION TITLE
How to prevent page scrolling when floating element is opened

### DISCUSSION THREAD

**Discussion by sweinstein22** [floating-ui/floating-ui#2680] [Q&A] [answered]
Hi everyone,

It seems like floating-ui scrolls the page for the user whenever my menus are opened, which is a more jarring UX than I'm looking for. Does anyone have insight on why it might cause this scrolling and how to prevent the behavior?

My floating-ui setup:

```
  const [showMenu, setShowMenu] = useState(false);
  const { refs, floatingStyles, context } = useFloating({
    open: showMenu,
    onOpenChange: setShowMenu,
    placement: "bottom-end",
    whileElementsMounted: autoUpdate,
    middleware: [offset({ crossAxis: -1 })],
  });
  const dismiss = useDismiss(context);
  const { getReferenceProps, getFloatingProps } = useInteractions([dismiss]);
```

A video of what I'm seeing:

https://github.com/floating-ui/floating-ui/assets/21115276/09d8bd68-e652-4901-bdc7-47b51ebdf770

Thanks so much for your help!

**Comment by atomiks** [comment]
It looks like you need to add `flip()` in your middleware array, after `offset`. The browser is scrolling the focused item (what it looks like at least) since it's partially offscreen when it's opening.

**Reply by sweinstein22** [reply]
I am actually interested in keeping the floating element at the bottom of the button to open it, I'm okay with it staying off screen until the user scrolls down, and in adding `flip()` interestingly it doesn't actually seem to prevent the scrolling!

https://github.com/floating-ui/floating-ui/assets/21115276/f6492ec7-c676-4403-8014-f5ef9a0c9e6a

**Reply by atomiks** [reply, marked as answer]
Can you share the full code/reproduction? How are you focusing the item? `.focus({preventScroll: true})` should work if you're manually doing it than using the lib's features 🤔

**Reply by sweinstein22** [reply]
Ooh that did it! I didn't realize we were manually adding focus in another part of the component. Thank you so much for the help!
